runtime/js-compute-runtime/builtins/request-response.cpp

```diff
@@ -29,8 +29,68 @@
 namespace builtins {
 
 namespace {
+bool error_stream_controller_with_pending_exception(JSContext *cx, JS::HandleObject controller) {
+  JS::RootedValue exn(cx);
+  if (!JS_GetPendingException(cx, &exn))
+    return false;
+  JS_ClearPendingException(cx);
+
+  JS::RootedValueArray<1> args(cx);
+  args[0].set(exn);
+  JS::RootedValue r(cx);
+  return JS::Call(cx, controller, "error", args, &r);
+}
 constexpr size_t HANDLE_READ_CHUNK_SIZE = 8192;
 
+bool process_body_read(JSContext *cx, int32_t handle, JS::HandleObject context,
```

Comment:
This used to be within event_loop.cpp but I moved it into request-response as it is only used on requests